docs/architecture/ErrorReporting.md

```diff
@@ -0,0 +1,140 @@
+                                        ignored.         ◄───────────────  sendMessage()   ◄────────────────── REST call
+                                                                           returns failed promise              fails
+                                                                                    │
+                                                                            &&      │
+                                                                                    ▼
+                                                                           updates ChatState.errors
+                                                                                    │
+                                                                                    │
+                                                                                    ▼
+                                    adaptedErrorBarSelector ◄───────────── onStateChange() event
+                                        extracts error                         with new error
+                                        and updates error bar UI
+                                                │
+                                     &&         │
+                                                ▼
+                                    updates adapter.state.errors
+                                                │
+                                     &&         │
+                                                ▼
+    Custom error handling◄────────── triggers on('error', handler)
+    logic
+
+* `SendBox.sendMessageOnClick()` handler runs when the user clicks on a button, outside Reach component lifecycle.
+* The asynchronously returned error is dropped by the handler.
+* But the error is also persisted by the stateful client.
+  * The `ErrorBar` gets updated based on the state update.
+  * The `ChatComposite` also updates its own internal state and triggers an event that the application acts on.
+
+## UI errors
+
+These errors are _unexpected_ in that the UI library has a bug or is unable to handle an error from the browser. Neither the end-user, nor the surrounding application, is expected to act on the errors. As such, they aren't handled via the mechanism used for ACS backend errors.
+
+The recommended way to handle these errors is to use React `ErrorBoundary`.
+
+    TODO: Add storybook example of using `ErrorBoundary`
+
+### Browser compatibility
+
+A special case of UI errors are errors due to browser incompatibility. The UI library does not determine browser compatibility at runtime. Any errors arising on unsupported browser (versions) are considered UI errors. The surrounding application can decide how to handle browser compatibility - e.g. by checking and warning end-users about browser versions or using polyfills where applicable. e.g. The samples included in this repository include a browser compatibility check.
```

Comment:
Re the existing sample browser errors - are these the ones that check if mobile and if not firefox? If so I aren't those ACS specific errors so are a category 1 error instead of special case? What are other examples of this special case?

Reply:
Yeah this is about those browser checks.

Category 1 errors aren't ACS specific errors. They're errors returned by the ACS backends.

The errors here really stem from whatever limitations the JS headless SDKs + our UI library has. So these are still category 2 errors - we'll probably get js errors due to this incompatibility.

I reworded this to make it explicit that I'm only talking about the case when you're using the ACS SDKs.

Reply:
I'm still confused on the definitions sorry! This error isn't because the browser didn't implement the ACS SDK, the "unsupported browser" error is an error from the ACS SDK (even though under the hood it might be due to browser limitation) - so by definition would fall into category 1 as I get the error when calling the ACS SDK?

Reply:
Then for the composite section it says "All errors are teed to an `errors` field". When referring to these errors, are they cat1 errors? (though they might not be from an ACS SDK?)

Reply:
Dropped the confusing classification of errors. Important point is errors that arise from the ACS clients are handled by the stateful client. Similarly, errors that arise behind the adapter API are expected to be handled by the adapters.

PTAL.